policies/versioning-policy.md

```diff
@@ -0,0 +1,155 @@
+# OpenSSL release versioning policy
+
+This document describes the release versioning scheme used by the OpenSSL
+project from version 3.0.0 onwards.  It also details the level of [ABI] and
+[API] compatibility each version represents.
+
+> _Note: All examples herein are illustrative and do not constitute part of the
+> versioning policy._
+
+The version scheme consists a triple of numbers:
+**[major]**.**[minor]**.**[patch]**.
+
+> For example, the version _3.0.1_ has a _major_ version of _3_, a _minor_
+> version of _0_ and a _patch_ version of _1_.
+
+This closely aligns with the expectations of users who are familiar
+with _semantic versioning_.  However, it is not quite _semantic versioning_
+because adopting _semantic versioning_ would mean changing our current
+Long Term Support ([LTS]) policies and practices.
+
+## Major release: changing the first digit
+
+A major release can, and generally will, introduce API and ABI breaking
+changes.
+
+> For example, a program that runs with OpenSSL 3.0.1 should not expect to run
+> with OpenSSL 4.0.0 without modifications.  If it does run, it might not be
+> as efficient as it was.
+
+In order to maintain stability and limit rework across major versions:
+
+- Existing [public interfaces] will remain unmodified except where changes
+  are unlikely to break source compatibility or where structures are being
+  made opaque.
+- Existing public interfaces must remain in at least one LTS release
+  until the original interface has been documented as _deprecated_
+  for at least 5 years.
```

Comment:
We seem to have lost wording about how long the replacement needs to have been available. We seem to swap between referring to "existing public interfaces" and "original interface"...are they referring to the same thing in this context?

Reply:
The original says:
> No existing public interface can be removed until its replacement has been in place in an LTS stable release.

I think this is covered by _Existing public interfaces must remain in at least one LTS release ..._.

Reply:
Well not really. In theory at least the time that we deprecate and the time that we implement a replacement may not be the same time. So, in theory, you could deprecate an interface in an LTS release but implement its replacement in the following non-LTS release.

Reply:
We could of course require that any replacement must be implemented in the same release which would also address the issue.

Reply:
I've reverted to the original wording.